Pr/entity iterator fixes

KuceraMartin opened the pull request.

hrach commented:
Thanks for the first Orm contribution! 🎉 
I wonder how's that it you are changing tests and everything still works. :( 
Is it ready to merge?

KuceraMartin commented:
I added the `once()` everywhere because by default any number of calls (0 - inf.) is accepted and I removed line 38 because that method is not supposed to be called at all (hasProperty() returns false)

I also rearranged the order of the mock expectancies definitions to match the order of method calls in EntityIterator.

I think it's ready to merge.

hrach commented:
I see, thanks for explanation, I missed that. But you bug is still not fixes, am I right?
thanks for the PR.
